HW14/proxy/txt_proxy.py

```diff
@@ -0,0 +1,34 @@
+from txt_reader import TxtReader
+from txt_writer import TxtWriter
+
+
+class TxtProxyReaderAndWriter:
+
+    def __init__(self, txt_reader: TxtReader, txt_writer: TxtWriter):
+        self.__result = ''
+        self.__is_actual = False
+        self.__reader = txt_reader
+        self.__writer = txt_writer
+
+    def read_file(self):
+        if self.__is_actual:
+            return self.__result
+        else:
+            self.__result = self.__reader.read_file()
+            self.__is_actual = True
+            return self.__result
+
+    def write_file(self, user_data):
+        if user_data == self.__result:
+            self.__is_actual = False
```

Comment:
No! in this case just do Nthng. Why do you change actual if you do not change data